src/main/java/esa/sen2vm/utils/BandInfo.java

```diff
@@ -0,0 +1,205 @@
+
+    /**
+     * Band name
+     */
+    protected String name = null;
+
+    /**
+     * Band index
+     */
+    protected int index = 0;
+    /**
+     * Band pixel height
+     */
+    protected double pixelHeight = 0;
+
+    /**
+     * Private constructor
+     * @param name band name
+     * @param pixelHeight band pixel size in meter)
+     */
+    private BandInfo(String name, int index, double pixelHeight) {
+        this.name = name;
+        this.index = index;
+        this.pixelHeight = pixelHeight;
+    }
+
+    /**
+     * Get BandInfo from band name
+     * @param bandName band name
+     * @return the band having the given name. Null if not found
+     */
+    public static BandInfo getBandInfoFromName(String bandName) {
+        for (BandInfo band : BandInfo.values()) {
+            if (band.name.equals(bandName)) {
+                return band;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Get BandInfo from sensor name
+     * @param sensorName sensor name
+     * @return the band having the given name. Null if not found
+     */
+    public static BandInfo getBandInfoFromSensorName(String sensorName) {
+        int underscoreIndex = sensorName.indexOf('_');
+        String bandName = sensorName.substring(1, underscoreIndex);
+        return getBandInfoFromName(bandName);
+    }
+
+    /**
+     * Get BandInfo from band name
+     * @param bandName band name
+     * @return the band having the given name. Null if not found
+     */
+    public static BandInfo getBandInfoFromNameWithB(String bandName) {
+        for (BandInfo band : BandInfo.values()) {
+            String name = "B" + band.getName2Digit();
+            if (name.equals(bandName)) {
+                return band;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Get BandInfo from band index
+     * @param bandIndex band index (from 0 to 12)
+     * @return the band having the given index (from 0 to 12). Null if not found
+     */
+    public static BandInfo getBandInfoFromIndex(int bandIndex) {
+        int nbBand = BandInfo.values().length;
+        if (bandIndex < 0 || bandIndex >= nbBand) {
+            return null;
+        }
+        return BandInfo.values()[bandIndex];
+    }
+
+    /**
+     * @return the index
+     */
+    public int getIndex() {
+        return index;
+    }
+
+    /**
+     * @return the name
+     */
+    public String getName() {
+        return name;
+    }
+
+    /**
+     * @return the name
+     */
+    public String getName2Digit() {
+        if (name.length() == 1) {
+            return "0" + name;
+        }
+        return name;
+    }
+
+    /**
+     * @return the pixel height
+     */
+    public double getPixelHeight() {
+        return pixelHeight;
+    }
+
+    /**
+     * Return the number of line for a granule of this band
+     * @return the number of line for a granule of this band
+     */
+    public double getGranuleNbLine() {
```

Comment:
Is this needed here or in the granule class ?

Reply:
Done here: https://github.com/sen2vm/sen2vm-core/commit/6365dea45094dc8eaaa439d211322f625e249e26